### PR TITLE
Download to temp-file before moving to original path

### PR DESCRIPTION
**Pull request by nakulkar-msft**
- Downloads will happen to a temp file named `.azDownload-<jobID>-fileName`, in the same directory where file is being downloaded, before being moved to `fileName`.
- Partial/failed transfers will be cleaned as before.
- This will be default behaviour for downloads. Zero byte files, downloads to NULL and directories are not impacted.
- If files exist with similar name, they'll be overwritten. (But given we include jobID in name, if such files exist, that is user trying to spite AzCopy.)
- Name clashes may happen in certain cases described [here](https://github.com/Azure/azure-storage-azcopy/blob/630b37f8c8554f5b96d000dabfcb9342a9ad31c8/ste/mgr-JobPartTransferMgr.go#L368). This behaviour is retained.

### CODE REVIEW

File: ste/mgr-JobPartTransferMgr.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ type IJobPartTransferMgr interface {
 }
 
 type TransferInfo struct {
+	JobID 				   common.JobID
```
Comment (zezha-msft) [inline ste/mgr-JobPartTransferMgr.go:93]: If I understood correctly, duplicating the jobID info here would not bloat the plan files, but what happens if there's a resume?

**Review by zezha-msft — COMMENTED** [review]
Lgtm, please take a look at the resume scenario, where I think it may break.

File: ste/mgr-JobPartTransferMgr.go (hunk 1 above)
Comment (nakulkar-msft) [reply ste/mgr-JobPartTransferMgr.go:93]: This is a in-memory struct created from plan file. It'll be recreated on resume. Nothing from here is flushed to disk.

**Review by mohsha-msft — APPROVED** [review]
LGTM.

**Review by zezha-msft — APPROVED** [review]
:shipit:

**Pull request merged**